pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -1379,13 +1381,12 @@ func (c *PodAffinityChecker) getMatchingAntiAffinityTopologyPairsOfPods(pod *v1.
 	for _, existingPod := range existingPods {
 		existingPodNode, err := c.info.GetNodeInfo(existingPod.Spec.NodeName)
 		if err != nil {
```

Comment:
doing this is fine and simple because all the pods are from the same node.

```suggestion
		if err != nil || existingPodNode == nil {
```